hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractDeleteHandler.java

```diff
@@ -87,9 +88,13 @@ public void handle(@NonNull final HandleContext context) throws HandleException
         final var accountStore = context.readableStore(ReadableAccountStore.class);
         final var toBeDeleted = requireNonNull(accountStore.getContractById(op.contractIDOrThrow()));
         final var obtainer = getObtainer(accountStore, op);
-        if (obtainer == null || obtainer.deleted()) {
+
+        validateTrue(obtainer != null, OBTAINER_DOES_NOT_EXIST);
```

Comment:
:warning: It seems what you want here is,
```
validateTrue(obtainer != null, OBTAINER_DOES_NOT_EXIST);
if (obtainer.deleted()) {
    throw new HandleException(obtainer.smartContract() ? INVALID_CONTRACT_ID : OBTAINER_DOES_NOT_EXIST);
}
```
...because we want to **always** fail on a deleted obtainer, including when that obtainer is a deleted smart contract.